### PR TITLE
Command display name

### LINKED ISSUE #1692: Command Display Name
Add support to override the name displayed on the UI with a new property for `display_name` in the command annotation. Then if it is present, utilize that on any pages displaying commands.

### PR DESCRIPTION
**Pull request by 1maple1**
fixes: #1692 
brewtils: command_display_name

### CODE REVIEW

File: src/app/beer_garden/api/http/handlers/v1/request.py [hunk 1]
```diff
@@ -372,7 +370,6 @@ async def get(self):
 
         # Add the filter for only requests the user is permitted to see
         q_filter = self.permitted_objects_filter(Request)
-        query_args["q_filter"] = q_filter
```
Comment (TheBurchLog) [inline src/app/beer_garden/api/http/handlers/v1/request.py]: ?

**Comment by TheBurchLog** [comment]
We should update the Authorization logic to support display name as well:
https://github.com/beer-garden/beer-garden/blob/develop/src/app/beer_garden/authorization.py#L209

**Comment by TheBurchLog** [comment]
Also the JS for role creation: 
https://github.com/beer-garden/beer-garden/blob/develop/src/ui/src/js/controllers/admin_role.js#L447

**Comment by TheBurchLog** [comment]
In pre_save for Requests, we should populate the `display_name` with `command` if null:
https://github.com/beer-garden/beer-garden/blob/develop/src/app/beer_garden/db/mongo/models.py#L461

**Comment by TheBurchLog** [comment]
We should update this function to be just `contains_field`:
https://github.com/beer-garden/beer-garden/blob/develop/src/app/beer_garden/db/mongo/util.py#L130

Then for the migration script you could do:
```python

if not contains_field("request",["command_display_name"]):
    request_collection = db.get_collection("request")
    for legacy_request in request_collection .find():
        legacy_request.command_display_name = legacy_request.command
        legacy_request.save()
        
```
Or if the index get automatically added prior to this, you will need to pop off the first record from the database to see if it is populated, if it isn't, then do the migration for all of the records.

File: src/app/beer_garden/api/http/handlers/v1/request.py (hunk 1 above)
Comment (1maple1) [reply src/app/beer_garden/api/http/handlers/v1/request.py]: done

**Comment by TheBurchLog** [comment]
Need to also add command display name for the outbound checks:
https://github.com/beer-garden/beer-garden/blob/command_display_name/src/app/beer_garden/authorization.py#L537
https://github.com/beer-garden/beer-garden/blob/command_display_name/src/app/beer_garden/authorization.py#L581

Could either update `_check` to support `command_name` and `command_display_name`, or run the check a second time with the display name.

File: src/app/beer_garden/db/mongo/util.py [hunk 2]
```diff
@@ -257,13 +257,23 @@ def ensure_v3_27_model_migration():
         db.drop_collection("legacy_role")
 
 
+def ensure_v3_29_model_migration():
+    db = get_db()
+    if not contains_field("request", ["command_display_name"]):
+        request_collection = db.get_collection("request")
+        for legacy_request in request_collection.find():
```
Comment (TheBurchLog) [inline src/app/beer_garden/db/mongo/util.py:269]: Update this to the following code.
``` python
legacy_request['command_display_name'] = legacy_request['command']
request_collection.update_one({"_id":legacy_request["_id"]}, {"$set":legacy_request})
```

File: src/app/beer_garden/db/mongo/util.py [hunk 3]
```diff
@@ -257,13 +257,23 @@ def ensure_v3_27_model_migration():
         db.drop_collection("legacy_role")
 
 
+def ensure_v3_29_model_migration():
+    db = get_db()
+    if not contains_field("request", ["command_display_name"]):
+        request_collection = db.get_collection("request")
```
Comment (TheBurchLog) [inline src/app/beer_garden/db/mongo/util.py:268]: Add Warning statement like the other migration scripts to let the Admins know you are updating their database.

**Review by TheBurchLog — APPROVED** [review]

**Pull request merged**